x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -206,7 +207,18 @@ public void accept(ClusterState state) {
                     for (Map<String, Object> processorConfigWithKey : processorConfigs) {
                         for (Map.Entry<String, Object> entry : processorConfigWithKey.entrySet()) {
                             if (TYPE.equals(entry.getKey())) {
-                                count++;
+                                ++count;
+                            }
+                            // Special handling as `foreach` processors allow a `processor` to be defined
+                            if (FOREACH_PROCESSOR_NAME.equals(entry.getKey())) {
+                                if (entry.getValue() instanceof Map<?, ?>) {
+                                    Object processorDefinition = ((Map<?, ?>)entry.getValue()).get("processor");
+                                    if (processorDefinition instanceof Map<?, ?>) {
+                                        if (((Map<?, ?>) processorDefinition).keySet().contains(TYPE)) {
+                                            ++count;
+                                        }
+                                    }
+                                }
```

Comment:
The for-each processor and the onFailure directive are the only scenarios I know of that result in child processors. Both of those can be nested to an indefinite number of levels. I'm not sure how far you want to go down that rabbit hole, though.

Reply:
Yeesh, yeah, thats right. 

Ugh, I will do the recursion. But who on earth would have a foreach processor nested in a foreach processor :(

Reply:
Yep, hence the 🐇 hole. 😃

Reply:
Only allow recursion up to 10 layers. Handling `on_failure` and `foreach`. seems to work ok :D

Reply:
Looks good to me. Thanks for making that change.